### PR TITLE
feature: ada supply query

### PR DESCRIPTION
**Pull request by rhyslbw**
# Context
It's currently not possible to get the complete supply information for ADA from genesis and on-chain data solely, and in addition, hitting the database to calculate the UTxO set balance is a costly operation so is unsuitable for exposing publicly.

# Proposed Solution
Adds a top-level query to return the principle currency supply information,
including circulating, max supply
(read from protocol params), and total supply (max - reserve balance).
cardano-cli query ledger-state is used to access the reserve balance,
with the result piped through jq in a sub-process. API requests are served
using the cache. This query can be safely to exposed to public consumers

# Important Changes Introduced
- The Dockerfile now builds `cardano-cli`, and is based on `ubuntu:20.04`
- Test mode defaults to `e2e`, using the simplified docker stack configuration

### CODE REVIEW

File: nix/nixos/cardano-graphql-service.nix [hunk 1]
```diff
@@ -53,6 +53,11 @@ in {
         default = "allegra";
       };
       
+      logLevel = lib.mkOption {
```
Comment (rhyslbw) [inline nix/nixos/cardano-graphql-service.nix:56]: @jbgi You can set this to `debug` for more information relating to startup

File: packages/api-cardano-db-hasura/src/CardanoNodeClient.ts [hunk 2]
```diff
@@ -39,7 +40,10 @@ export class CardanoNodeClient {
       await fs.stat(process.env.CARDANO_NODE_SOCKET_PATH)
       const { slotNo } = await this.cardanoCli.getTip()
       if (slotNo < this.currentEraFirstSlot) {
-        throw new Error('cardano-node is still synchronizing')
+        this.logger.debug('cardano-node tip', { module: 'CardanoNodeClient', value: slotNo })
```
Comment (rhyslbw) [inline packages/api-cardano-db-hasura/src/CardanoNodeClient.ts:43]: @jbgi

File: nix/nixos/cardano-graphql-service.nix [hunk 3]
```diff
@@ -126,12 +145,17 @@ in {
...
         CARDANO_NODE_SOCKET_PATH = cfg.cardanoNodeSocketPath;
+        CURRENT_ERA_FIRST_SLOT = cfg.currentEraFirstSlot;
+        ERA_NAME = cfg.eraName;
         POOL_METADATA_PROXY = cfg.smashUrl;
         GENESIS_FILE_BYRON = cfg.genesisByron;
         GENESIS_FILE_SHELLEY = cfg.genesisShelley;
         HASURA_CLI_PATH = hasura-cli + "/bin/hasura";
         HASURA_URI = hasuraBaseUri;
+        JQ_PATH = pkgs.jq + "/bin/jq";
+        LOGGER_LEVEL = cfg.loggerLevel;
```
Comment (jbgi) [inline nix/nixos/cardano-graphql-service.nix:158]:
```suggestion
        LOGGER_LEVEL = cfg.logLevel;
```

File: nix/nixos/cardano-graphql-service.nix [hunk 4]
```diff
@@ -126,12 +145,17 @@ in {
       wantedBy = [ "multi-user.target" ];
       requires = [ "graphql-engine.service" ];
       environment = lib.filterAttrs (k: v: v != null) {
+        CARDANO_CLI_PATH = cfg.cardanoCliPackage + "/bin/cardano-cli";
         CARDANO_NODE_SOCKET_PATH = cfg.cardanoNodeSocketPath;
+        CURRENT_ERA_FIRST_SLOT = cfg.currentEraFirstSlot;
```
Comment (jbgi) [inline nix/nixos/cardano-graphql-service.nix:150]:
```suggestion
        CURRENT_ERA_FIRST_SLOT = toString cfg.currentEraFirstSlot;
```

**Review by jbgi — APPROVED** [review]

**Pull request merged**